### ISSUE TITLE
Started getting "invalid name" errors on a module-level global variable after upgrade to pylint 4.0.4

### ISSUE THREAD

**Issue by baruchoxman** [PyCQA/pylint#10768]
### Bug description

```python
pylint --include-naming-hint=y utils/myfile.py

utils/myfile.py:16:0: C0103: Constant name "_g_is_installed" doesn't conform to UPPER_CASE naming style ('([^\\W\\da-z][^\\Wa-z]*|__.*__)$' pattern) (invalid-name)
```

This is the content - the global is changed later on hence it is not a const:
```
_g_is_installed = False


def x():
    global _g_is_installed
    _g_is_installed = True
```

### Configuration

```ini

```

### Command used

```shell
pylint --include-naming-hint=y myfile.py
```

### Pylint output

```python
myfile.py:16:0: C0103: Constant name "_g_is_installed" doesn't conform to UPPER_CASE naming style ('([^\\W\\da-z][^\\Wa-z]*|__.*__)$' pattern) (invalid-name)
```

### Expected behavior

Before the upgrade it was passing

### Pylint version

```shell
python 3.12
pylint 4.0.4
astroid 4.0.2
```

### Additional dependencies

```python

```

**Comment by DanielNoord** [comment]
This was also discussed in https://github.com/pylint-dev/pylint/issues/10700

I think we concluded that we indeed consider this a bug, but I'm not 100% anymore...

**Comment by ThosRTanner** [comment]
I've had this start as well. Up to 4.0.3 is fine, but 4.0.4 produces the warning about invalid constant names. This is a non backward compatible change which seems to serve no purpose. Even, better

```python
fred = somefunction()

def myfunc():
    global fred
    fred = True
```

will sometimes give the warning and sometimes not! e.g.

```python
import random

def wibble():
    return random.choice((True,))

fred = wibble()

def myfunc():
    global fred
    fred = True
```

gives no warning about the name `fred`, but

```python
def wibble():
    return True

fred = wibble()

def myfunc():
    global fred
    fred = True
```

produces a warning.

I mean it's nice that you can do `logger = logging.getLogger()` without getting the warning, but this behaviour is *incredibly* confusing for the end user.

Also discovered after reading through the mentioned issue, that I can use `__fred__` to get round this but this feels wrong. It certainly goes against the advise in PEP8
